bigtable/client/client_options.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef BIGTABLE_CLIENT_CLIENTOPTIONS_H_
+#define BIGTABLE_CLIENT_CLIENTOPTIONS_H_
+
+#include <google/bigtable/v2/bigtable.grpc.pb.h>
+#include <grpc++/grpc++.h>
+#include <grpc++/support/channel_arguments.h>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+class ClientOptions {
+ public:
+  ClientOptions();
+  const std::string& endpoint() const { return endpoint_; }
+  ClientOptions& SetEndpoint(const std::string& endpoint) {
+    endpoint_ = endpoint;
+    return *this;
+  }
+  std::shared_ptr<grpc::ChannelCredentials> credentials() const {
+    return credentials_;
+  }
+  ClientOptions& SetCredentials(
+      std::shared_ptr<grpc::ChannelCredentials> credentials) {
+    credentials_ = credentials;
+    return *this;
+  }
+  const grpc::ChannelArguments channel_arguments() const { return channel_arguments_; }
```

Comment:
I think we will want separate setter / getter arguments for each channel argument (e.g., `SetLoadBalancingPolicyName()`, and `SetCompressionAlgorithm()`), no need to fix that in this PR, just add a issue for it ("Consider adding setter / getter for each channel argument").

Reply:
Created new Issue.